src/Form/ReactionRuleAddForm.php

```diff
@@ -71,9 +72,18 @@ public function form(array $form, FormStateInterface $form_state) {
       '#title' => $this->t('React on event'),
       '#options' => $options,
       '#required' => TRUE,
+      '#ajax' => $this->getDefaultAjax(),
       '#description' => $this->t('Whenever the event occurs, rule evaluation is triggered.'),
```

Comment:
Let's add executes submit callback here as in \Drupal\Core\Render\Element\Submit and add       '#submit' => array('::submitForm'), as \Drupal\Core\Entity\EntityForm::actions , this should update the entity in form state correctly.